api/src/services/lottery.service.ts

```diff
@@ -1113,31 +1132,71 @@ export class LotteryService {
       throw new ForbiddenException();
     }
 
-    const applicationUserId = await this.prisma.applications.findFirstOrThrow({
+    if (!user.userRoles.isAdmin) {
```

Comment:
Public users don't have userRoles attached to them. So when I go through the flow with a public user I created this throws a null pointer exception